app/queues/dns/dns-record-worker.server.ts

```diff
@@ -0,0 +1,211 @@
+import { UnrecoverableError, Worker } from 'bullmq';
+import { RecordStatus } from '@prisma/client';
+
+import { redis } from '~/lib/redis.server';
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { createRecord, upsertRecord, deleteRecord, getChangeStatus } from '~/lib/dns.server';
+
+import type { Record } from '@prisma/client';
+
+export const dbRecordUpdateQueueName = 'db-record-update';
+export const dnsRecordUpdateQueueName = 'dns-record-update';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export enum WorkType {
+  create = 'create',
+  update = 'update',
+  delete = 'delete',
+}
+export interface JobRecord {
+  workType?: WorkType;
+  id?: Record['id'];
+  username: Record['username'];
+  type: Record['type'];
+  name: Record['name'];
+  value: Record['value'];
+}
+
+export type addDbRecordJobRecordResult = Record['id'];
+export type dnsRecordUpdateJobRecordResult = string;
+
+export const dbRecordWorker = new Worker<JobRecord>(
+  dbRecordUpdateQueueName,
+  async (job) => {
+    logger.debug('DB record update in DB');
+    const { workType, id, username, type, name, value } = job.data;
+
+    try {
+      switch (workType) {
+        case 'create':
+          const existingRecords = await prisma.record.count({
```

Comment:
Is there any way we can collapse all this code into a function you call in your switch, like you do in 104-116 below?  This is pretty hard to read with it all expanded inline like this.

Reply:
Will be updated!